turtle_graphics/turtle.js

```diff
@@ -0,0 +1,130 @@
+class Turtle {
+  constructor(initialX, initialY) {
+    this.x = initialX;
+    this.y = initialY;
+    this.path = [[this.x, this.y]];
+    this.direction = "x";
+    this.maxY = -Infinity;
+    this.minY = Infinity;
+    this.maxX = -Infinity;
+    this.minX = Infinity;
+  }
+
+  forward(steps) {
+    for (let i = 0; i < steps; i++) {
+      if (this.direction == "x") {
+        this.x++;
+      } else if (this.direction === "-y") {
+          this.y--;
+      } else if (this.direction === "-x") {
+        this.x--;
+      } else {
+        this.y++;
+      }
+      this.path.push([this.x, this.y]);
+    }
+    return this;
+  }
+
+  right() {
+    switch (this.direction) {
+      case "x": 
+        this.direction = "-y";
+        break;
+      case "y": 
+        this.direction = "x";
+        break;
+      case "-x": 
+        this.direction = "y";
+        break;
+      case "-y": 
+        this.direction = "-x";
+        break;
+      default: 
+        return this;
+      } 
+    return this;
+  }
+
+  left() {
+    switch (this.direction) {
+      case "x": 
+        this.direction = "y";
+        break;
+      case "-y": 
+        this.direction = "x";
+        break;
+      case "-x": 
+        this.direction = "-y";
+        break;
+      case "y": 
+        this.direction = "-x";
+        break;
+      default: 
+        return this;
+    }
+    return this;
+  }
+
+  canvasSize() {
+    for (let position of this.path) {
+      if (position[1] > this.maxY) {
+        this.maxY = position[1];
+      }
+    }
+    for (let position of this.path) {
+      if (position[1] < this.minY) {
+        this.minY = position[1];
+      }
+    }
+    for (let position of this.path) {
+      if (position[0] > this.maxX) {
+        this.maxX = position[0];
+      }
+    }
+    for (let position of this.path) {
+      if (position[0] < this.minX) {
+        this.minX = position[0];
+      }
+    }
```

Comment:
I just realized you're actually looping 4 times over the same array. You can achieve this in a single pass:
```javascript
for (let position of this.path) {
  if (position[1] > this.maxY) {
    this.maxY = position[1];
  }
  if (position[1] < this.minY) {
    this.minY = position[1];
  }
  // ...
}
```
Also, as you might have seen in an earlier comment, another way to achieve each of these is via `Math.min()` and `Math.max()`
```javascript
for (let position of this.path) {
  this.maxX = Math.max(this.maxX, position.x);
  this.minX = Math.min(this.minX, position.x);
  this.maxY = Math.max(this.maxY, position.y);
  this.minY = Math.min(this.maxY, position.y);
}
```
Also, note the anal-retention by putting the X stuff before the Y stuff, since that's how you normally think of coordinates (x,y, not y,x)